### PR TITLE
bmon: Add basic alerting logic

### PR DESCRIPTION
**Pull request by maurelian**
**Description**

A `src` dir with a basic monitor that queries balances but does not yet emit any alerts is added.

**Tests**

A test is added that succeeds when balances are healthy and no findings are reported.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1100,3 +1106,5 @@ workflows:
             - op-batcher-docker-publish
           context:
             - gcr-release
+
+# VS Code Extension Version: 1.5.0
```
Comment (trianglesphere) [inline .circleci/config.yml:1110]: Could you remove this line?

Comment (maurelian) [reply .circleci/config.yml:1110]: Yep. It's a great extension, why it wants to do this IDK.

**Comment by maurelian** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4556** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/4556" target="_blank">Graphite</a>  👈
    * **PR #4560** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/4560" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/4556?utm_source=stack-comment).

**Comment by maurelian** [comment]
The Docker and CI config were reviewed as part of an upstack branch, which was merged into this branch in #4560.

File: packages/balance-monitor/src/agent.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      const accountBalance = BigNumber.from(
+        (
+          await provider.getBalance(account.address, blockEvent.blockNumber)
+        ).toString()
+      )
+      if (accountBalance.gte(account.thresholds.warning)) {
+        // return if this is the last account
+        if (idx === accounts.length - 1) {
+          return findings
+        } else {
```
Comment (mslipper) [inline packages/balance-monitor/src/agent.ts:50]: You won’t need this continue if you’re returning above

File: packages/balance-monitor/src/agent.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+          continue
+        }
+      }
+    }
+
+    // todo: Add to the findings array when balances are below the threshold
+
+    return findings
```
Comment (mslipper) [inline packages/balance-monitor/src/agent.ts:58]: Is adding to the findings array what triggers the alert?

**Review by mslipper — APPROVED** [review]
have some nits + a question about functionality otherwise LGTM.

File: packages/balance-monitor/src/agent.ts (hunk 3 above)
Comment (maurelian) [reply packages/balance-monitor/src/agent.ts:58]: Not added yet to keep the PR small.

**Comment by maurelian** [comment]
Subsequent PRs will: 
1. Hard code know addresses: 
	2. Mainnet and Goerli accounts imported from contracts package
	3. Beta accounts hard-coded into this package
4. Add the ops genie key/alert creation endpoint to the `.env` file, and create alerts: 
	5. P3 for balances below the 'warning' threshold
	6. P1 for balances below the 'danger' threshold

**Comment by maurelian** [comment]
@mergifyio requeue

**Pull request merged**